File: cloud/acp-admin-cloud-dependencies/src/main/java/pers/acp/admin/common/permission/oauth/RuntimeConfigExpression.java
package pers.acp.admin.common.permission.oauth;

import pers.acp.admin.common.constant.ModuleFuncCode;
import pers.acp.admin.common.constant.RoleCode;
import pers.acp.admin.common.permission.BaseExpression;

/**
 * 定义运行参数配置权限表达式
 *
 * @author zhang by 28/12/2018
 * @since JDK 11
 */
public interface RuntimeConfigExpression extends BaseExpression {

    /**
     * 运行参数配置
     */
    String runtimeConfig = "hasAnyAuthority('" + RoleCode.prefix + RoleCode.ADMIN + "','" + ModuleFuncCode.runtimeConfig + "')";

    /**
     * 运行参数新增
     */
    String runtimeAdd = "hasAnyAuthority('" + RoleCode.prefix + RoleCode.ADMIN + "','" + ModuleFuncCode.runtimeAdd + "')";

    /**
     * 运行参数删除
     */
    String runtimeDelete = "hasAnyAuthority('" + RoleCode.prefix + RoleCode.ADMIN + "','" + ModuleFuncCode.runtimeDelete + "')";

    /**
     * 运行参数更新
     */
    String runtimeUpdate = "hasAnyAuthority('" + RoleCode.prefix + RoleCode.ADMIN + "','" + ModuleFuncCode.runtimeUpdate + "')";

    /**
     * 运行参数查询
     */
    String runtimeQuery = "hasAnyAuthority('" + RoleCode.prefix + RoleCode.ADMIN + "','" + ModuleFuncCode.runtimeQuery + "')";

}

File: cloud/oauth-server/src/main/kotlin/pers/acp/admin/oauth/constant/RuntimeConfigExpression.kt
package pers.acp.admin.oauth.constant

import pers.acp.admin.constant.ModuleFuncCode
import pers.acp.admin.constant.RoleCode
import pers.acp.admin.permission.BaseExpression

/**
 * 定义运行参数配置权限表达式
 *
 * @author zhang by 28/12/2018
 * @since JDK 11
 */
object RuntimeConfigExpression : BaseExpression() {
    const val superOnly = BaseExpression.superOnly
    const val sysConfig = BaseExpression.sysConfig
    /**
     * 运行参数配置
     */
    const val runtimeConfig = "hasAnyAuthority('" + RoleCode.prefix + RoleCode.SUPER + "','" + ModuleFuncCode.runtimeConfig + "')"

    /**
     * 运行参数新增
     */
    const val runtimeAdd = "hasAnyAuthority('" + RoleCode.prefix + RoleCode.SUPER + "','" + ModuleFuncCode.runtimeAdd + "')"

    /**
     * 运行参数删除
     */
    const val runtimeDelete = "hasAnyAuthority('" + RoleCode.prefix + RoleCode.SUPER + "','" + ModuleFuncCode.runtimeDelete + "')"

    /**
     * 运行参数更新
     */
    const val runtimeUpdate = "hasAnyAuthority('" + RoleCode.prefix + RoleCode.SUPER + "','" + ModuleFuncCode.runtimeUpdate + "')"

    /**
     * 运行参数查询
     */
    const val runtimeQuery = "hasAnyAuthority('" + RoleCode.prefix + RoleCode.SUPER + "','" + ModuleFuncCode.runtimeQuery + "')"
}